src/photos/ducks/clustering/albums.js

```diff
@@ -37,24 +42,113 @@ const createAutoAlbum = async (albums, photos) => {
   }
 }
 
-export const findAllAutoAlbums = async () => {
-  const albums = await cozyClient.data.findAll(DOCTYPE_ALBUMS)
-  return albums.filter(album => album.auto)
+export const findAutoAlbums = async () => {
+  const autoAlbums = await cozyClient.data.defineIndex(DOCTYPE_ALBUMS, [
+    'auto',
+    'name'
+  ])
+  const results = await cozyClient.data.query(autoAlbums, {
+    selector: { auto: true },
+    sort: [{ name: 'asc' }]
+  })
+  return results
 }
 
-// TODO: deal with updates
-export const saveClustering = async clusters => {
-  const albums = await findAllAutoAlbums()
+export const saveClustering = async (clusters, albums) => {
   for (const photos of clusters) {
     if (photos && photos.length > 0) {
-      const album = await createAutoAlbum(albums, photos)
-      const refs = await createReferences(album, photos)
-      if (refs) {
-        log(
-          'info',
-          `${photos.length} photos clustered into: ${JSON.stringify(album)}`
-        )
+      const album = await createAutoAlbum(photos, albums)
+      await createReferences(photos, album)
+    }
+  }
+}
+
+const findPhotosByAlbum = async album => {
+  album._type = DOCTYPE_ALBUMS
+  const files = await cozyClient.data.fetchReferencedFiles(album, {})
+  if (files && files.included) {
+    const attributes = files.included.map(file => {
+      const attributes = file.attributes
+      attributes.id = file.id
+      return attributes
+    })
+    return extractPhotosInfo(attributes)
+  }
+  return []
+}
+
+const findPhotosToClusterize = async albums => {
```

Comment:
Yeah I had a hard time for the naming :/
In this case, we are trying to find the photos already clusterized for these albums, because at least one new photo (not clusterized) match this cluster by date ; so we need to recompute the whole cluster.
Maybe `findPhotosToReclusterize` would be better?